src/app-layer-protos.c

```diff
@@ -101,10 +64,44 @@ AppProto StringToAppProto(const char *proto_name)
         return ALPROTO_UNKNOWN;
 
     // We could use a Multi Pattern Matcher
-    for (size_t i = 0; i < ARRAY_SIZE(AppProtoStrings); i++) {
+    for (size_t i = 0; i < ALPROTO_MAX; i++) {
         if (strcmp(proto_name, AppProtoStrings[i].str) == 0)
             return AppProtoStrings[i].alproto;
     }
 
     return ALPROTO_UNKNOWN;
 }
+
+void RegisterAppProtoString(AppProto alproto, const char *proto_name)
```

Comment:
can ALPROTO_FAILED just be a constant instead of having it be a dynamic value for no real reason I can see?

Reply:
ALPROTO_FAILED is used in comparison like `< ALPROTO_FAILED` ...
Will think about it again

Reply:
Looks like we can move  ALPROTO_FAILED to 1 and change the comparisons... in next PR version